sdk/webpubsub/Microsoft.Azure.WebJobs.Extensions.WebPubSub/src/Config/WebPubSubConfigProvider.cs

```diff
@@ -0,0 +1,152 @@
+        }
+
+        public Task<HttpResponseMessage> ConvertAsync(HttpRequestMessage input, CancellationToken cancellationToken)
+        {
+            return _dispatcher.ExecuteAsync(input, _options.AllowedHosts, _options.AccessKeys, cancellationToken);
+        }
+
+        private void ValidateWebPubSubConnectionAttributeBinding(WebPubSubConnectionAttribute attribute, Type type)
+        {
+            ValidateConnectionString(
+                attribute.ConnectionStringSetting,
+                $"{nameof(WebPubSubConnectionAttribute)}.{nameof(WebPubSubConnectionAttribute.ConnectionStringSetting)}");
+        }
+
+        private void ValidateWebPubSubAttributeBinding(WebPubSubAttribute attribute, Type type)
+        {
+            ValidateConnectionString(
+                attribute.ConnectionStringSetting,
+                $"{nameof(WebPubSubAttribute)}.{nameof(WebPubSubAttribute.ConnectionStringSetting)}");
+        }
+
+        internal WebPubSubService GetService(WebPubSubAttribute attribute)
+        {
+            var connectionString = Utilities.FirstOrDefault(attribute.ConnectionStringSetting, _options.ConnectionString);
+            var hubName = Utilities.FirstOrDefault(attribute.Hub, _options.Hub);
+            return new WebPubSubService(connectionString, hubName);
+        }
+
+        private IAsyncCollector<WebPubSubOperation> CreateCollector(WebPubSubAttribute attribute)
+        {
+            return new WebPubSubAsyncCollector(GetService(attribute));
+        }
+
+        private WebPubSubConnection GetClientConnection(WebPubSubConnectionAttribute attribute)
+        {
+            var hub = Utilities.FirstOrDefault(attribute.Hub, _options.Hub);
+            var service = new WebPubSubService(attribute.ConnectionStringSetting, hub);
+            return service.GetClientConnection(attribute.UserId);
+        }
+
+        private void ValidateConnectionString(string attributeConnectionString, string attributeConnectionStringName)
+        {
+            AddSettings(attributeConnectionString);
+            var connectionString = Utilities.FirstOrDefault(attributeConnectionString, _options.ConnectionString);
+
+            if (string.IsNullOrEmpty(connectionString))
+            {
+                throw new InvalidOperationException($"The Service connection string must be set either via an '{Constants.WebPubSubConnectionStringName}' app setting, via an '{Constants.WebPubSubConnectionStringName}' environment variable, or {attributeConnectionStringName}.");
+            }
+        }
+
+        private void AddSettings(string connectionString)
+        {
+            if (!string.IsNullOrEmpty(connectionString))
+            {
+                var item = new ServiceConfigParser(connectionString);
+                _options.AllowedHosts.Add(item.Endpoint.Host);
+                _options.AccessKeys.Add(item.AccessKey);
+            }
+        }
+
+        private static T ConvertWebPubSubOperationFromJObject<T>(JObject input)
```

Comment:
try `input.ToObject` and pass in a JsonSerializer with WebPubSubOperationJsonConverter?